parlai/chat_service/services/websocket/websocket_manager.py

```diff
@@ -9,234 +9,68 @@
-        while time_passed < 5 * 60:
-            any_alive = False
-            for task_fut in self.active_worlds.values():
-                if task_fut is not None:
-                    any_alive = any_alive or task_fut.running()
-            if not any_alive:
-                break
-            time.sleep(1)
-
-        # Tell the worlds that they should be shutting down at this point,
-        # agents will refer to this value
-        self.shutting_down = True
-
-    def _get_unique_pool(self):
-        """Return unique pool.
-
-        Returns a filtered version of the agent pool where each agent is
-        only listed a maximum of one time.
-
-        :return:
-            a dictionary mapping world_types to agent pools
-        """
-        valid_pools = {}
-        for world_type, agent_pool in self.agent_pool.items():
-            eligibility_function = shared_utils.get_eligibility_fn(
-                self.world_module, world_type
-            )
-            if eligibility_function is not None:
-                valid_pools[world_type] = [
-                    w for w in agent_pool if eligibility_function(w)
-                ]
-            else:
-                valid_pools[world_type] = self.agent_pool[world_type]
-        return valid_pools
-
-    def get_agent_state(self, agent_id):
-        """Return agent state.
-
-        :param agent_id:
-            int agent identifier
 
-        :return:
-            AgentState object if agent_id is being tracked, else None
-        """
-        if agent_id in self.messenger_agent_states:
-            return self.messenger_agent_states[agent_id]
-        return None
+        self.message_sender = WebsocketManager.MessageSender()
 
-    def _get_agent(self, agent_id, task_id):
-        """Return agent object for given agent ID and task ID.
+        self.service_reference_id = None
 
-        :param agent_id:
-            int agent identifier
-        :param task_id:
-            string task name
+        self._parse_config(opt)
+        self._complete_setup()
 
-        :return:
-            MessengerAgent object associated with given agent ID and task ID if
-            possible, else None
-        """
-        agent_state = self.get_agent_state(agent_id)
-        if agent_state is not None:
-            if agent_state.has_task(task_id):
-                return agent_state.get_agent_for_task(task_id)
-        return None
+    def parse_additional_args(self, opt):
+        pass
 
     def _complete_setup(self):
         """Complete necessary setup items."""
+        self.agent_pool = {}
+        self.messenger_agent_states = {}
+        self.agent_id_to_overworld_future = {}
+        self.task_to_agent_ids = {}
         self._load_model()
 
     def _load_model(self):
         """Load model if necessary"""
         if 'model_file' in self.opt or 'model' in self.opt:
             self.opt['shared_bot_params'] = create_agent(self.opt).share()
 
+    def _handle_message_read(self, event):
```

Comment:
what is this for? can you add a docstring?

Reply:
Added a docstring